src/main/java/com/dragonballs/responsedatamapping/UserLocationCreator.java

```diff
@@ -0,0 +1,17 @@
+package com.dragonballs.responsedatamapping;
+
+import com.dragonballs.entities.User;
+import org.springframework.stereotype.Service;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import java.net.URI;
+
+@Service
+public class UserLocationCreator {
+
+    public URI userLocationCreator(User user){
+        URI location = ServletUriComponentsBuilder.fromCurrentRequest().path("/{id}")
```

Comment:
Small thingy, but you could simply write one line:
return ServletUriComponentsBuilder.fromCurrentRequest().path("/{id}")
                .buildAndExpand(user.getId()).toUri();